File: src/EflayGameSaveManager.Core/Models/AppSnapshot.cs
namespace EflayGameSaveManager.Core.Models;

public sealed record AppSnapshot(
    string ConfigPath,
    string BackupRoot,
    IReadOnlyList<GameSnapshot> Games,
    IReadOnlyDictionary<string, DeviceDefinition> Devices,
    CurrentDeviceContext CurrentDevice);

public sealed record GameSnapshot(
    string Name,
    bool CloudSyncEnabled,
    IReadOnlyList<ResolvedSaveUnit> SaveUnits,
    IReadOnlyList<ResolvedGamePath> GamePaths,
    IReadOnlyList<CurrentDevicePathInfo> CurrentDevicePaths,
    CurrentDeviceGamePathInfo CurrentDeviceGamePath);

public sealed record ResolvedSaveUnit(
    int Id,
    SaveUnitType UnitType,
    bool DeleteBeforeApply,
    IReadOnlyList<ResolvedDevicePath> Paths);

public sealed record ResolvedGamePath(string DeviceId, string DeviceName, string Path);

public sealed record ResolvedDevicePath(string DeviceId, string DeviceName, string Path);

File: src/EflayGameSaveManager.Kotlin/app/src/main/java/com/eflay/gamesavemanager/model/AppSnapshot.kt
package com.eflay.gamesavemanager.model

data class AppSnapshot(
    val games: List<GameSnapshot>,
    val currentDevice: CurrentDeviceContext
)

data class GameSnapshot(
    val name: String,
    val cloudSyncEnabled: Boolean,
    val saveUnits: List<SaveUnitSnapshot>
)

data class SaveUnitSnapshot(
    val id: Int,
    val unitType: SaveUnitType,
    val path: String?,
    val deleteBeforeApply: Boolean = false
)

data class CurrentDeviceContext(
    val deviceId: String,
    val deviceName: String
)
